executor/ddl.go

```diff
@@ -791,20 +791,35 @@ func (e *DDLExec) executeFlashbackTable(s *ast.FlashBackTableStmt) error {
 }
 
 func (e *DDLExec) executeLockTables(s *ast.LockTablesStmt) error {
+	if !config.TableLockEnabled() {
+		noopFuncsMode := e.ctx.GetSessionVars().NoopFuncsMode
+		err := ErrFuncNotEnabled.GenWithStackByArgs("Lock tables", "enable-table-lock")
+		if noopFuncsMode == variable.OffInt {
+			return err
+		} else if noopFuncsMode == variable.WarnInt {
+			e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+		}
+		return nil
```

Comment:
Sorry, for this syntax I think we can only `WARN` at most. I know this is different from most other noop functions, but this syntax is used by `mysqldump`. If we generate an error, mysqldump files won't restore by default :(

Reply:
The behavior will be very strange. If we only have `WARN`, I think use `TableLockEnabled()` is better than `noopFuncsMode `.

Reply:
I agree it is strange, but the change is too severe if mysqldump can not be restored.

For the docs: https://docs.pingcap.com/tidb/dev/system-variables#tidb_enable_noop_functions-new-in-v40

We can add a sentence: Setting `tidb_enable_noop_functions` to `OFF` does not prevent `LOCK TABLE t1 WRITE` from executing. A warning is returned instead. This behavior is required for compatibility with mysqldump.

Reply:
Updated, PTAL @morgo